gtsam/slam/SmartProjectionRigFactor.h

```diff
@@ -0,0 +1,366 @@
+ * L. Carlone, Z. Kira, C. Beall, V. Indelman, F. Dellaert, Eliminating conditionally
+ * independent sets in factor graphs: a unifying perspective based on smart factors,
+ * Int. Conf. on Robotics and Automation (ICRA), 2014.
+ */
+
+/**
+ * This factor assumes that camera calibration is fixed (but each measurement
+ * can be taken by a different camera in the rig, hence can have a different
+ * extrinsic and intrinsic calibration). The factor only constrains poses (variable dimension
+ * is 6 for each pose). This factor requires that values contains the involved poses (Pose3).
+ * If all measurements share the same calibration (i.e., are from the same camera), use SmartProjectionPoseFactor instead!
+ * If the calibration should be optimized, as well, use SmartProjectionFactor instead!
+ * @addtogroup SLAM
+ */
+template<class CAMERA>
+class SmartProjectionRigFactor : public SmartProjectionFactor<CAMERA> {
+
+ private:
+  typedef SmartProjectionFactor<CAMERA> Base;
+  typedef SmartProjectionRigFactor<CAMERA> This;
+  typedef typename CAMERA::CalibrationType CALIBRATION;
+
+  static const int DimPose = 6;  ///< Pose3 dimension
+  static const int ZDim = 2;  ///< Measurement dimension
+
+ protected:
+
+  /// vector of keys (one for each observation) with potentially repeated keys
+  KeyVector nonUniqueKeys_;
+
+  /// cameras in the rig (fixed poses wrt body + fixed intrinsics, for each camera)
+  typename Base::Cameras cameraRig_;
+
+  /// vector of camera Ids (one for each observation, in the same order), identifying which camera took the measurement
+  FastVector<size_t> cameraIds_;
+
+ public:
+  typedef CAMERA Camera;
+  typedef CameraSet<CAMERA> Cameras;
+
+  /// shorthand for a smart pointer to a factor
+  typedef boost::shared_ptr<This> shared_ptr;
+
+  /// Default constructor, only for serialization
+  SmartProjectionRigFactor() {
+  }
+
+  /**
+   * Constructor
+   * @param sharedNoiseModel isotropic noise model for the 2D feature measurements
+   * @param cameraRig set of cameras (fixed poses wrt body and intrinsics) in the camera rig
+   * @param params parameters for the smart projection factors
+   */
+  SmartProjectionRigFactor(const SharedNoiseModel& sharedNoiseModel,
+                           const Cameras& cameraRig,
+                           const SmartProjectionParams& params =
+                               SmartProjectionParams())
+      : Base(sharedNoiseModel, params),
+        cameraRig_(cameraRig) {
+    // use only configuration that works with this factor
+    Base::params_.degeneracyMode = gtsam::ZERO_ON_DEGENERACY;
+    Base::params_.linearizationMode = gtsam::HESSIAN;
+  }
+
+  /**
+   * Constructor
+   * @param sharedNoiseModel isotropic noise model for the 2D feature measurements
+   * @param camera single camera (fixed poses wrt body and intrinsics)
+   * @param params parameters for the smart projection factors
+   */
+  SmartProjectionRigFactor(const SharedNoiseModel& sharedNoiseModel,
```

Comment:
I don't understand why we have this constructor, but if you must, delegate to constructor above using { camera}